ksqldb-rest-app/src/main/java/io/confluent/ksql/api/auth/JaasAuthProvider.java

```diff
@@ -44,40 +42,34 @@
  */
 public class JaasAuthProvider implements AuthProvider {
 
-  private static final Logger log = LoggerFactory.getLogger(JaasAuthProvider.class);
+  private static final Logger LOG = LoggerFactory.getLogger(JaasAuthProvider.class);
 
   private final Server server;
-  private final KsqlRestConfig config;
-  private final LoginContextSupplier loginContextSupplier;
-  private final List<String> allowedRoles;
   private final String contextName;
+  private final LoginContextSupplier loginContext;
 
-  public JaasAuthProvider(final Server server, final KsqlRestConfig config) {
-    this(server, config, JAASLoginService::new);
+  @VisibleForTesting
+  @FunctionalInterface
+  interface LoginContextSupplier {
+    JAASLoginService get();
+  }
+
+  public JaasAuthProvider(
+      final Server server,
+      final String contextName
+  ) {
+    this(server, contextName, JAASLoginService::new);
   }
 
   @VisibleForTesting
   JaasAuthProvider(
       final Server server,
-      final KsqlRestConfig config,
+      final String contextName,
       final LoginContextSupplier loginContextSupplier
   ) {
     this.server = Objects.requireNonNull(server, "server");
-    this.config = Objects.requireNonNull(config, "config");
-    this.loginContextSupplier =
-        Objects.requireNonNull(loginContextSupplier, "loginContextSupplier");
-    final List<String> authRoles = config.getList(KsqlRestConfig.AUTHENTICATION_ROLES_CONFIG);
-    this.allowedRoles = authRoles.stream()
-        .filter(role -> !"*".equals(role)) // remove "*"
-        .map(role -> "**".equals(role) ? "*" : role) // Change "**" to "*"
-        .collect(Collectors.toList());
```

Comment:
this was moved to a dedicated class because authorization and authentication were split into two in VertX 4 (see https://vert-x3.github.io/vertx-4-migration-guide/index.html#authentication-and-authorization_{context})